File: frontend/src/types/container.ts
/**
 * Container related types
 */

export interface DevContainerResponse {
  status: 'not_created' | 'created' | 'running' | 'stopped' | 'exited';
  container_id?: string;
  dev_url?: string;
}

export interface DevServerStatus {
  running: boolean;
  port?: number;
  pid?: number;
  uptime?: number;
}

export interface ProdContainerResponse {
  status: 'not_created' | 'created' | 'running' | 'stopped' | 'exited';
  container_id?: string;
  prod_url?: string;
}

export interface ProdBuildResponse {
  success: boolean;
  message: string;
  image_id?: string;
}

File: src/mgx_api/schemas/container.py
"""Container related schemas."""
from pydantic import BaseModel


class DevContainerResponse(BaseModel):
    """Dev container status response."""
    container_id: str | None = None
    container_name: str | None = None
    status: str
    ports: dict[str, str] = {}
    dev_url: str | None = None


class ProdBuildResponse(BaseModel):
    """Production build response."""
    image_id: str | None = None
    image_tag: str | None = None
    status: str
    logs: str | None = None
    error: str | None = None


class ProdContainerResponse(BaseModel):
    """Production container status response."""
    container_id: str | None = None
    container_name: str | None = None
    status: str
    ports: dict[str, str] = {}
    prod_url: str | None = None
